src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs

```diff
@@ -91,15 +95,22 @@ public int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler e
                 // Use "ToArray" to copy ValueColleciton to a simple array and use it's enumerator
                 // Set the enumerator for parallel yielding of testCases
                 // Whenever a concurrent executor becomes free, it picks up the next set of testCases using this enumerator
-                this.testCaseListEnumerator = testCasesBySource.Values.ToArray().GetEnumerator();
+                var testCaseLists = testCasesBySource.Values.ToArray();
+                this.testCaseListEnumerator = testCaseLists.GetEnumerator();
+                this.availableTestSources = testCaseLists.Length;
             }
             else
             {
                 // Set the enumerator for parallel yielding of sources
                 // Whenever a concurrent executor becomes free, it picks up the next source using this enumerator
                 this.sourceEnumerator = testRunCriteria.Sources.GetEnumerator();
+                this.availableTestSources = testRunCriteria.Sources.Count();
```

Comment:
I am fine with name but if you are changing here then please change in ParallelProxyDiscoveryManager.cs also